packages/cli/create-strapi-app/src/types.ts

```diff
@@ -32,3 +38,57 @@ export type DBConfig = {
     ssl?: boolean;
   };
 };
+
+export type PackageManager = 'npm' | 'yarn' | 'pnpm';
+
+export interface Scope {
+  name: string;
+  rootPath: string;
+  template?: string;
+  templateBranch?: string;
+  templatePath?: string;
+  strapiVersion?: string;
+  installDependencies?: boolean;
+  devDependencies?: Record<string, string>;
+  dependencies?: Record<string, string>;
+  docker?: boolean;
+  packageManager: PackageManager;
+  runApp?: boolean;
+  isQuickstart?: boolean;
+  uuid?: string;
+  deviceId?: string;
+  database: DatabaseInfo;
+  tmpPath?: string;
+  packageJsonStrapi?: Record<string, unknown>;
+  useTypescript?: boolean;
+  useExampleApp?: boolean;
+  gitInit?: boolean;
+}
+
+export type ClientName = 'mysql' | 'postgres' | 'sqlite';
+
+export interface DatabaseInfo {
+  client: ClientName;
+  connection?: {
+    host?: string;
+    port?: string;
+    database?: string;
+    username?: string;
+    password?: string;
+    filename?: string;
+    ssl?: boolean;
+  };
+}
+
+export interface StderrError extends Error {
+  stderr: string;
+}
+
+export function isStderrError(error: unknown): error is StderrError {
```

Comment:
Couldn't we create an StderrError instead and do error instanceof StderrError?

Reply:
we are not the ones throwing the errors so we would still need this test to rewrap on a new class that's not really necessary